README: update with information about RB3G2 and compose-apps (#2)
foundriesio/meta-partner-variscite: README.md

# Qualcomm FoundriesFactory Reference Manual

## Table of Contents

> - Document Revision History

Reviewer: I think we can add links to the sections below like:
```
[Document Revision History](#document-revision-history)
```

But this needs to be tested

Reviewer 2: you can view the rendered version here https://github.com/foundriesio/meta-partner/blob/c4168e5d0a3c1aeabb98f5f9250afb1c18700a52/README.md there's a couple of odd formatting things.

Author: @mike-scott I'll try to add the link anchors.

  - [Deploy gst-ai-object-detection Application](#deploy-gst-ai-object-detection-application)
  - [Deploy gst-ai-parallel-inference Application](#deploy-gst-ai-parallel-inference-application)
  - [Deploy gst-ai-pose-detection Application](#deploy-gst-ai-pose-detection-application)
  - [Deploy gst-ai-segmentation Application](#deploy-gst-ai-segmentation-application)
  - [Deploy gst-multi-camera-example Application](#deploy-gst-multi-camera-example-application)
  - [Enable/Disable Wireguard VPN](#enabledisable-wireguard-vpn)
- [FoundriesFactory Images](#foundriesfactory-images)
  - [LmP Factory Image](#lmp-factory-image)
- [Wireguard VPN](#wireguard-vpn)
- [Updating FoundriesFactory Images](#updating-foundriesfactory-images)
- [Useful Links](#updating-foundriesfactory-images)

## Document Revision History

| Version | Date       | Comments        |
|---------|------------|-----------------|
| v0.1    | 2024-07-01 | Initial release |

## Definitions, Acronyms, and Abbreviations

| Variable        | Meaning                                                           |
|-----------------|-------------------------------------------------------------------|

> | `<factory>`     | Your factory name                                                 |

Reviewer 3: ```suggestion
| `<factory>`     | Your Factory name                                                 |
```

| `<device_name>` | Device name used when you registered the device                   |
| `<app_name>`    | Docker-compose app available to run on the device                 |
| `<tag_name>`    | The branch you want your device to follow                         |
| `<IP>`          | Device IP                                                         |

## Getting Started

> To get started using the FoundriesFactory and the Linux microPlatform, please follow the steps below:

Reviewer 3: ```suggestion
To get started using the FoundriesFactory™ Platform and the Linux microPlatform, please follow the steps below:
```
If we don't want to use "Platform" as it is followed by "microPlatform", the alternatives, in preferred order are:
* Sass
* Service
* Software
* Solution

1. To start a new Factory, please visit https://app.foundries.io/ and sign up for an account.
2. In the Platform drop down, select Qualcomm RB3 Gen 2 Development Kit.

> 3. Give your factory a name, and click "Create Factory".

Reviewer 3: ```suggestion
3. Give your Factory a name, and click "Create Factory".
```

> 4. The factory owners may invite additional users to their FoundriesFactory via email under the "Members" tab in the factory.

Reviewer 3: ```suggestion
4. The Factory owners may invite additional users to their FoundriesFactory Account via email under the "Members" tab of the Factory interface.
```

>     - Manage your factory users: https://app.foundries.io/factories/<factory>/members/

Reviewer 3: ```suggestion
    - Manage your Factory users: https://app.foundries.io/factories/<factory>/members/
```

5. Fioctl CLI Installation: https://docs.foundries.io/latest/getting-started/install-fioctl/index.html

> 6. Configure git using fioctl: https://docs.foundries.io/latest/getting-started/git-config/index.html

Reviewer 3: ```suggestion
6. Configure git using Fioctl: https://docs.foundries.io/latest/getting-started/git-config/index.html
```

**Note**: On macOS, you may encounter authentication issues due to Git on OSX using the Keychain Access Utility. The solution is to remove Keychain Access entries from your git config file.

> ---

Reviewer 3: ```suggestion

```

### Qualcomm® Robotics RB3G2 Development Kit

> Once your factory has been created, it will build the source code for the RB3G2 and produce a target. A target is a secure over-the-air update but also provides the build artifacts for initial provisioning.

Reviewer 3: ```suggestion
Once your Factory has been created, it will build the source code for the RB3G2 and produce a Target. A Target is a secure over-the-air update but also provides the build artifacts for initial provisioning.
```

1. Download the image: https://app.foundries.io/factories/<factory>/targets/1/artifacts/qcm6490/lmp-factory-image-qcm6490.qcomflash.tar.gz

> 2. Extract the tar.gz into a known location.

Reviewer 3: ```suggestion
2. Extract the `tar.gz` into a known location.
```

3. Open a terminal and change the directory into `lmp-factory-image-qcm6490`.
4. The compressed archive contains the flashing tool “qdl”.
   - Note: The tool from the build has the interpreter set incorrectly.

> 5. Download and compile qdl tool for your platform:

Reviewer 3: ```suggestion
5. Download and compile qdl for your platform:
```

   - `git clone https://github.com/linux-msm/qdl`
   - Read the README and install build dependencies
   - `cd qdl`
   - `make`
6. Disable ModemManager on some host systems if necessary.

Now configure the Qualcomm® Robotics RB3G2 Development Kit:

1. Set up DIP_SW_0 positions 1 and 2 to ON. This enables serial output to the debug port.
2. Connect the USB debug cable to the host. Baud rate is 115200.
   - You can use your favorite UART client to access the console, such as minicom, putty etc.
3. The serial connection is based on the FTDI chip:
   - `/dev/serial/by-id/usb-FTDI_FT230X_Basic_UART_<serial ID>-if00-port0`
4. Plug in the USB-C cable from the host.
5. Run the command to flash:
   - `./qdl --debug prog_firehose_ddr.elf rawprogram*.xml patch*.xml`
6. Press and hold the F_DL button and connect the power cable.

> ---

Reviewer 3: ```suggestion

```

### Serial Console

After flashing, the device should boot the Linux microPlatform properly. Use your serial console to log in.

Username: `fio`
Password: `fio`

> ---

Reviewer 3: ```suggestion

```

### Connect via WiFi

Using the WiFi radio, connect it to the internet using NetworkManager CLI:

```
nmcli device wifi connect “<AP Name>” password “<AP password>”
```

The sudo password is `fio`.

> ---

Reviewer 3: ```suggestion

```

### Connect via Ethernet

In order to enabled Ethernet, you must provide firmware in the yocto recipe below. This is a one time operation, which will also enable USB type A ports to function.

Register and log in to https://www.renesas.com, then download firmware from
https://www.renesas.com/us/en/products/interface/usb-switches-hubs/upd720201-usb-30-host-controller.

Once downloaded, copy USB3-201-202-FW-20131112.zip at recipes-firmware/firmware/renesas-upd720201
and add the renesas-upd720201 package to your image.

#### Add downloaded firmware zip to the correct layer location

> ```bash

Reviewer 3: ```suggestion

```bash
```

git clone https://source.foundries.io/factories/<factory>/meta-subscriber-overrides.git
cd meta-subscriber-overrides
mkdir -p recipes-firmware/firmware/renesas-upd720201
cp /tmp/USB3-201-202-FW-20131112.zip recipes-firmware/firmware/renesas-upd720201
echo 'FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"' > recipes-firmware/firmware/renesas-upd720201_20131112.bbappend
echo 'SRC_URI += "file://USB3-201-202-FW-20131112.zip"' >> recipes-firmware/firmware/renesas-upd720201_20131112.bbappend
```

#### Add renesas-upd720201 to lmp-factory-image

> ```bash

Reviewer 3: ```suggestion

```bash
```

echo 'CORE_IMAGE_BASE_INSTALL += "renesas-upd720201"' >> recipes-samples/images/lmp-factory-image.bb
```

#### Commit and push to create a new build
```bash
git add .
git commit -s -m "renesas-upd720201: add firmware file"
git push
```

After you push, the FoundriesFactory will build a new target. Once built, any registered device will update over the air.

> ---

Reviewer 3: ```suggestion

```

### SSH

Once connected to the network you can log in over ssh if desired:

```bash
ssh fio@qcm6490.local
```

OR

```bash
ssh fio@<IP>
```

Password: `fio`

---

> ### Register your device

Reviewer 3: ```suggestion
### Register Your Device
```

For more information, please see the project's [README](https://github.com/foundriesio/qimsdk-lmp/blob/main/README.md).

### gst-concurrent-videoplay-composition

The gst-concurrent-videoplay-composition command-line application facilitates concurrent video decode and playback for advanced video coding (AVC)-coded videos and performs composition on a display for video wall application. The application requires at least one input video file, which is expected to be an MP4 file with the AVC codec.

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-concurrent-videoplay-composition.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-concurrent-videoplay-composition compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-concurrent-videoplay-composition.disabled gst-concurrent-videoplay-composition
git commit -s -m “gst-concurrent-videoplay-composition” && git push
```

> This compose-app requires a display connected for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display to function properly.
```

### gst-ai-classification

> The gst-ai-classification application enables you to recognize the subject in the image. The use cases use Qualcomm Neural Processing SDK runtime or TensorFlow Lite (TFLite) runtime. The compose file provides you will a few commands you can comment/uncomment to use different models for classification.

Reviewer 3: ```suggestion
The gst-ai-classification application enables subject recognition in the image. The use cases uses the Qualcomm Neural Processing SDK runtime or the TensorFlow Lite (TFLite) runtime. The compose file provides you will a few commands you can comment/uncomment to use different models for classification.
```

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-ai-classification.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-classification compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-classification.disabled gst-ai-classification
git commit -s -m “gst-ai-classification” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-ai-daisychain-detection-classification

The gst-ai-daisychain-detection-classification application enables you to perform cascaded object detection and classification with a camera and a file source. The use case involves detecting objects and classifying the detected objects.

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/daisy-chain-detection-and-classification.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-daisychain-detection-classification compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-daisychain-detection-classification.disabled gst-ai-daisychain-detection-classification
git commit -s -m “gst-ai-daisychain-detection-classification” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-ai-monodepth

The gst-ai-monodepth application enables you to infer depth from a live camera stream.

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/mono-depth-from-video.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-monodepth compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-monodepth.disabled gst-ai-monodepth
git commit -s -m “gst-ai-monodepth” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-ai-object-detection

The gst-ai-object-detection application enables you to detect objects within images and videos. The use cases demonstrate the execution of YOLOv5, YOLOv8, and YOLO-NAS using the Qualcomm Neural Processing SDK runtime.

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-ai-object-detection.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-object-detection compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-object-detection.disabled gst-ai-object-detection
git commit -s -m “gst-ai-object-detection” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-ai-parallel-inference

> The gst-ai-parallel-inference application enables you to perform object detection, object classification, pose detection, and image segmentation on a live camera stream. The use cases use Qualcomm Neural Processing SDK runtime for object detection and image segmentation, and TFLite runtime for classification and pose detection.

Reviewer 3: ```suggestion
The gst-ai-parallel-inference application enables you to perform object detection, object classification, pose detection, and image segmentation on a live camera stream. The use cases uses the Qualcomm Neural Processing SDK runtime for object detection and image segmentation, and the TFLite runtime for classification and pose detection.
```

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-ai-parallel-inference.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-parallel-inference compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-parallel-inference.disabled gst-ai-parallel-inference
git commit -s -m “gst-ai-parallel-inference” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-ai-pose-detection

> The gst-ai-pose-detection application enables you to detect the body pose of the subject in an image or video. The use cases use a video stream from a camera, leverage TFLite for pose detection, and display the results on the screen.

Reviewer 3: ```suggestion
The gst-ai-pose-detection application enables you to detect the body pose of the subject in an image or video. The use cases use a video stream from a camera, leverages TFLite for pose detection, and displays the results on the screen.
```

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-ai-pose-detection.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-pose-detection compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-pose-detection.disabled gst-ai-pose-detection
git commit -s -m “gst-ai-pose-detection” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-ai-segmentation

> The gst-ai-segmentation application enables you to divide an image into different and meaningful parts or segments and assign a label to each homogenous segment based on the similarity of the attributes. The application shows how to use Qualcomm Neural Processing SDK runtime and TFLite runtime for image segmentation.

Reviewer 3: ```suggestion
The gst-ai-segmentation application enables you to divide an image into meaningful parts or segments and assign a label to each homogenous segment based on similarity of the attributes.
The application shows how to use both the Qualcomm Neural Processing SDK runtime and TFLite runtime for image segmentation.
```

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-ai-segmentation.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-ai-segmentation compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-ai-segmentation.disabled gst-ai-segmentation
git commit -s -m “gst-ai-segmentation” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### gst-multi-camera-example

> The gst-multi-camera-example application enables you to simultaneously stream from two camera sensors on the device. The application composes camera feeds side-by-side to display on a screen or encodes and stores the video streams to files. Typical use cases that need multiple camera inputs are dash camera or stereo camera, which can use this application as a reference and build a use case.

Reviewer 3: ```suggestion
The gst-multi-camera-example application enables you to simultaneously stream from two camera sensors on the device. It composes camera feeds side-by-side to display on a screen, or encodes and stores the video streams to file. Typical use cases that need multiple camera inputs are dash camera or stereo camera, which can use this application as a reference to build a use case.
```

For a more detailed description please see [Qualcomm Linux Sample Apps](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-50/gst-multi-camera-stream-example.html?product=1601111740013072&facet=Qualcomm%20Intelligent%20Multimedia%20SDK).

Enable the gst-multi-camera-example compose app before deployment:

```bash
git clone https://source.foundries.io/factories/<factory>/containers.git
git mv gst-multi-camera-example.disabled gst-multi-camera-example
git commit -s -m “gst-multi-camera-example” && git push
```

> This compose-app requires a display connected, and the camera mezzanine for it to function properly.

Reviewer 3: ```suggestion
This compose-app requires a connected display and the camera mezzanine to function properly.
```

### Deploy

After you push, the FoundriesFactory will build a new target. Once it is built, you may deploy as described below.

> ---

Reviewer 3: ```suggestion

```

## Fioctl

> Fioctl is a command-line tool to manage your FoundriesFactory. To deploy applications listed below, you must have enabled them first as described above. The polling time on devices by default is five minutes, once you have issued your deployment command, please wait up to five minutes to see the change reflected.

Reviewer 3: ```suggestion
Fioctl is a command-line tool to manage your Factory. To deploy applications listed below, you must have enabled them first as described above. The polling time on devices by default is five minutes, once you have issued your deployment command, please wait up to five minutes to see the change reflected.
```

### Examples:

### View Targets:

>   ```bash
>   ./fioctl targets list -f <factory>
>   ```
> ### View Devices:
>   ```bash
>   ./fioctl devices list -f <factory>
>   ```
> ### Change Device Tag:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --tags <tag_name>
>   ```
> ### Change Device Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps <app_name1>,<app_name2>,<app_name3>
>   ```
> ### Deploy shellhttpd Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps shellhttpd
>   ```
> ### Deploy gst-concurrent-videoplay-composition Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-concurrent-videoplay-composition
>   ```
> ### Deploy gst-ai-classification Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-classification
>   ```
> ### Deploy gst-ai-daisychain-detection-classification Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-daisychain-detection-classification
>   ```
> ### Deploy gst-ai-monodepth Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-monodepth
>   ```
> ### Deploy gst-ai-object-detection Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-object-detection
>   ```
> ### Deploy gst-ai-parallel-inference Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-object-detection
>   ```
> ### Deploy gst-ai-pose-detection Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-pose-detection
>   ```
> ### Deploy gst-ai-segmentation Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-segmentation
>   ```
> ### Deploy gst-multi-camera-example Application:
>   ```bash
>   ./fioctl devices -f <factory> config updates <device_name> --apps gst-multi-camera-example
>   ```
> ### Enable/Disable Wireguard VPN:
>   ```bash
>   ./fioctl devices -f <factory> config wireguard <device_name> enable/disable
>   ```
> 

Reviewer 3: ```suggestion
#### View Targets:

  ```bash
  ./fioctl targets list -f <factory>
  ```

#### View Devices:

  ```bash
  ./fioctl devices list -f <factory>
  ```
#### Change Device Tag:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --tags <tag_name>
  ```

#### Change Device Application:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps <app_name1>,<app_name2>,<app_name3>
  ```

#### Deploy `shellhttpd`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps shellhttpd
  ```

#### Deploy `gst-concurrent-videoplay-composition`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-concurrent-videoplay-composition
  ```

#### Deploy `gst-ai-classification`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-classification
  ```

#### Deploy `gst-ai-daisychain-detection-classification`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-daisychain-detection-classification
  ```

#### Deploy `gst-ai-monodepth`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-monodepth
  ```

#### Deploy `gst-ai-object-detection`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-object-detection
  ```

#### Deploy `gst-ai-parallel-inference`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-object-detection
  ```

#### Deploy `gst-ai-pose-detection`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-pose-detection
  ```

#### Deploy `gst-ai-segmentation`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-ai-segmentation
  ```

#### Deploy `gst-multi-camera-example`:

  ```bash
  ./fioctl devices -f <factory> config updates <device_name> --apps gst-multi-camera-example
  ```

#### Enable/Disable Wireguard VPN:

  ```bash
  ./fioctl devices -f <factory> config wireguard <device_name> enable/disable
  ```

```

Reviewer 3: A different change to consider would be to factor things out for all the deploy application examples,reducing it to a single `./fioctl devices -f <factory> config updates <device_name> --apps <application_name>` which could be paired with a list or table of available apps.

Reviewer 3: I also do not know why the suggestion is being rendered rather than shown as part of a diff. I can submit a patch if GitHub is being weird.

Author: Yeah I can't add your suggest to a batch commit, If you could add a patch that would be great.

Reviewer 3: I am almost at the end of my usual workday, but if urgent I can either get it to you later today or this weekend. Otherwise I can do it first thing next week alongside anything else I catch.

Reviewer 3: Sent via email!

---

## FoundriesFactory Images

### LmP Factory Image

This image configuration includes all the required packages to be baked into the final image:
- `https://source.foundries.io/factories/<factory>/meta-subscriber-overrides.git/tree/recipes-samples/images/lmp-factory-image.bb?h=main`

---

## Wireguard VPN

Registered devices can form a VPN to allow remote access even behind firewalls:
- `https://docs.foundries.io/latest/reference-manual/remote-access/wireguard.html`

---

## Updating FoundriesFactory Images

> To update your factory, follow these steps:

Reviewer 3: ```suggestion
To update your Factory, follow these steps:
```

   - `git clone https://github.com/linux-msm/qdl`
   - Read the README and install build dependencies
   - `cd qdl`
   - `make`
6. Disable ModemManager on some host systems if necessary.

Now configure the Qualcomm® Robotics RB3G2 Development Kit:

1. Set up DIP_SW_0 positions 1 and 2 to ON. This enables serial output to the debug port.
2. Connect the USB debug cable to the host. Baud rate is 115200.
   - You can use your favorite UART client to access the console, such as minicom, putty etc.
3. The serial connection is based on the FTDI chip:
   - `/dev/serial/by-id/usb-FTDI_FT230X_Basic_UART_<serial ID>-if00-port0`
4. Plug in the USB-C cable from the host.
5. Run the command to flash:
   - `./qdl --debug prog_firehose_ddr.elf rawprogram*.xml patch*.xml`
6. Press and hold the F_DL button and connect the power cable.

### Serial Console

After flashing, the device should boot the Linux microPlatform properly. Use your serial console to log in.

Username: `fio`
Password: `fio`

### Connect via WiFi

Using the WiFi radio, connect it to the internet using NetworkManager CLI:

```
sudo nmcli device wifi connect “<AP Name>” password “<AP password>”
```

The sudo password is `fio`.

> ### Connect via Ethernet

Reviewer: Let's change this to something like: `Enabling Ethernet and USB`

Author: Done.

> In order to enabled Ethernet, you must provide firmware in the yocto recipe below. This is a one time operation, which will also enable USB type A ports to function.

Reviewer: - enabled -> enable
- capitalize Yocto

```In order to enable Ethernet, you must provide firmware in the Yocto recipe below. This is a one time operation, which will also enable USB type A ports to function.```

Author: Done.
